Marlin/src/module/stepper/trinamic.cpp

```diff
@@ -752,6 +756,60 @@ enum StealthIndex : uint8_t {
   }
 #endif // TMC2209
 
+#if HAS_DRIVER(TMC2240)
+  template<char AXIS_LETTER, char DRIVER_ID, AxisEnum AXIS_ID>
+  void tmc_init(TMCMarlin<TMC2240Stepper, AXIS_LETTER, DRIVER_ID, AXIS_ID> &st, const uint16_t mA, const uint16_t microsteps, const uint32_t hyb_thrs, const bool stealth, const chopper_timing_t &chop_init, const bool interpolate, float hold_multiplier) {
+    st.begin();
+    TMC2240_n::GCONF_t gconf{0};
+    gconf.en_pwm_mode = !stealth;
+    st.GCONF(gconf.sr);
+    st.en_pwm_mode(stealth);
+    st.stored.stealthChop_enabled = stealth;
+
+    TMC2240_n::DRV_CONF_t drv_conf{0};
+    st.Rref = TMC2240_Rref;
+    drv_conf.current_range = TMC2240_CURRENT_RANGE;
+    drv_conf.slope_control = TMC2240_SLOPE_CONTROL;
+    st.DRV_CONF(drv_conf.sr);
+
+    //SERIAL_ECHOLNPGM("mA=", mA);
+    st.rms_current(mA, hold_multiplier);
+
+    st.iholddelay(6);
+    st.irundelay(4);
+
+    TMC2240_n::CHOPCONF_t chopconf{0};
+    chopconf.toff = chop_init.toff;       //  3
+    chopconf.intpol = interpolate;        //  1
+    chopconf.hstrt = chop_init.hstrt - 1; //  6 - 1
+    chopconf.hend = chop_init.hend + 3;   // -1 + 3
+    chopconf.TBL    = 2;
+    chopconf.tpfd   = 4;
+    TERN_(EDGE_STEPPING, chopconf.dedge = true);
+
+    st.CHOPCONF(chopconf.sr);
+    st.microsteps(microsteps);
+    //st.CHOPCONF(0x14410153);  // 0x14410153
+
+    TMC2240_n::PWMCONF_t  pwmconf{0};
+    pwmconf.pwm_ofs = 30;
+    pwmconf.pwm_autoscale = 1;
+    pwmconf.pwm_autograd = 1;
+    pwmconf.pwm_reg = 4;
+    pwmconf.pwm_lim = 12;
+    st.PWMCONF(pwmconf.sr);
+    st.TPOWERDOWN(10);
+
+    //st.GCONF(0x00);
+    //st.IHOLD_IRUN(0x04071F03);
+    //st.CHOPCONF(0x14410153);  // 0x14410153
+    //st.PWMCONF(0xC40C1E1D);
+    st.GSTAT(0x07);
+    st.GSTAT(0x00);
+    delay(200);
```

Comment:
some here there is something that makes stealth chop not work well. Since using the 2240 as if it was a 2160 worked well, I think we can copy the reg values from it.
These are the only new settings I added:
```
    st.Rref = TMC2240_Rref;
    drv_conf.current_range = TMC2240_CURRENT_RANGE;
    drv_conf.slope_control = TMC2240_SLOPE_CONTROL;
    ```
    
    In discord another user reported it working well as a 5160 too (except for current and new settings of course)